ios/Scripts/upload_dsym_manual.sh

```diff
@@ -41,58 +74,132 @@ if [ "${#DSYM_TGZ_FILES[@]}" -eq 0 ]; then
   exit 1
 fi
 
-# Start building curl command
-CURL_COMMAND="curl --request PUT \
+DSYM_JSON_MAPPINGS="$DSYM_JSON_MAPPINGS ]"
+
+METADATA_FILE=$(mktemp)
+TEMP_FILES+=("$METADATA_FILE")
+
+JSON_PAYLOAD=$(cat <<-END_JSON
+{
+  "version_name": "$VERSION_NAME",
+  "version_code": "$VERSION_CODE",
+  "build_size": $BUILD_SIZE,
+  "build_type": "$BUILD_TYPE",
+  "app_unique_id": "$APP_UNIQUE_ID",
+  "os_name": "$OS_NAME",
+  "mappings": $DSYM_JSON_MAPPINGS
+}
+END_JSON
+)
+
+echo "$JSON_PAYLOAD" > "$METADATA_FILE"
+
+CURL_COMMAND_META="curl --request PUT \
   --url $API_URL/builds \
-  --header 'Authorization: Bearer $API_KEY'"
+  --header 'Authorization: Bearer $API_KEY' \
+  --header 'Content-Type: application/json'"
 
-# Append custom headers if provided
 IFS='|' read -r -a CUSTOM_HEADERS <<< "$RAW_CUSTOM_HEADERS"
 for HEADER in "${CUSTOM_HEADERS[@]}"; do
   if [ -n "$HEADER" ]; then
-    CURL_COMMAND="$CURL_COMMAND --header '$HEADER'"
+    CURL_COMMAND_META="$CURL_COMMAND_META --header '$HEADER'"
   fi
 done
 
-# Add base form fields
-CURL_COMMAND="$CURL_COMMAND \
-  --form version_name=$VERSION_NAME \
-  --form version_code=$VERSION_CODE \
-  --form build_size=$BUILD_SIZE \
-  --form build_type=$BUILD_TYPE \
-  --form app_unique_id=$APP_UNIQUE_ID \
-  --form os_name=$OS_NAME"
+CURL_COMMAND_META="$CURL_COMMAND_META --data @$METADATA_FILE"
+
+echo ""
+echo "Uploading Build Metadata..."
+RESPONSE_BODY_FILE=$(mktemp)
+TEMP_FILES+=("$RESPONSE_BODY_FILE")
+
+HTTP_STATUS_CODE=$(eval "$CURL_COMMAND_META --write-out '%{http_code}' --silent --output $RESPONSE_BODY_FILE")
+HTTP_RESPONSE_BODY=$(cat "$RESPONSE_BODY_FILE")
+
+if [[ "$HTTP_STATUS_CODE" -ne 200 && "$HTTP_STATUS_CODE" -ne 201 ]]; then
+    echo "[ERROR]: Metadata upload failed with status code $HTTP_STATUS_CODE."
+    exit 1
+fi
+
+echo ""
+echo "Uploading dSYM files..."
+UPLOAD_SUCCESS=true
+MAX_ATTEMPTS=3
+
+echo "$HTTP_RESPONSE_BODY" | jq -c '.mappings[]' | \
+while IFS= read -r URL_OBJECT; do
+    
+    SIGNED_URL=$(echo "$URL_OBJECT" | jq -r '.upload_url')
+    EXPECTED_FILENAME=$(echo "$URL_OBJECT" | jq -r '.filename')
+    
+    UPLOAD_HEADERS_CURL=""
+    HEADER_LINES=$(echo "$URL_OBJECT" | jq -r '.headers | to_entries[] | "\(.key): \(.value)"')
+
+    while IFS= read -r HEADER_LINE; do
+        if [ -n "$HEADER_LINE" ]; then
+            UPLOAD_HEADERS_CURL+=" --header \"$HEADER_LINE\""
+        fi
+    done <<< "$HEADER_LINES"
+    
+    if [ -z "$SIGNED_URL" ] || [ -z "$EXPECTED_FILENAME" ]; then
+        echo "[ERROR]: Failed to extract SIGNED_URL or FILENAME from JSON object. Aborting remaining uploads."
```

Comment:
"[ERROR]: Failed to read response from server. Stack traces will not be symbolicated."